.github/workflows/check-release-branch.yml

```diff
@@ -38,3 +38,55 @@ jobs:
                     echo "Base branch has commits since merge base"
                     exit 1
                   fi
+
+    check_pr_branch:
```

Comment:
felt like it made enough sense to have the checks in this job as separate steps, as opposed to creating separate dependent jobs for each check